client/app/bundles/KudosApp/components/KudosList.jsx

```diff
@@ -1,89 +1,68 @@
+import AppStore from '../stores/AppStore'
+import TabBar from '../components/TabBar'
+import Kudo from './Kudo'
 
 injectTapEventPlugin()
 
-const List = ({ userId, kudos, likeKudo, unlikeKudo, updateKudo, isFetchingKudos }) => (
-  <div className="kudos-list">
-    {kudos.length === 0 && !isFetchingKudos
-      ? 'No kudos'
-      : kudos.map(kudo => (
-          <Kudo
-            id={kudo.id}
-            colorClass={kudo.colorClass}
-            userId={userId}
-            key={kudo.id}
-            kudo={kudo}
-            likeKudo={likeKudo}
-            unlikeKudo={unlikeKudo}
-            updateKudo={updateKudo}
-          />
-        ))}
-  </div>
-)
-
-const Spinner = () => (
-  <div className="kudos-list__fetching-container">
-    <i className="fas fa-spin fa-spinner fa-5x" aria-hidden="true" />
-  </div>
-)
-
-export default class KudosList extends React.Component {
-  static propTypes = {
-    kudos: PropTypes.array.isRequired,
-    id: PropTypes.string.isRequired,
-    isFetchingKudos: PropTypes.bool.isRequired,
-    totalKudos: PropTypes.number.isRequired,
-    fetchPage: PropTypes.func.isRequired,
-    likeKudo: PropTypes.func.isRequired,
-    unlikeKudo: PropTypes.func.isRequired,
-    updateKudo: PropTypes.func.isRequired,
-  }
-
-  constructor(props, context) {
-    super(props, context)
-    _.bindAll(this, 'areMoreKudos', 'handleScroll')
-  }
-
-  areMoreKudos() {
-    return this.props.totalKudos > this.props.kudos.length
-  }
-
-  componentDidMount() {
-    window.addEventListener('scroll', this.handleScroll)
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener('scroll', this.handleScroll)
-  }
-
-  handleScroll(event) {
-    const isAtBottom =
-      event.srcElement.body.scrollTop + window.innerHeight == document.body.offsetHeight
-
-    if (isAtBottom && this.areMoreKudos() && !this.props.isFetchingKudos) {
-      this.props.fetchPage()
+@observer
+export class KudosList extends React.Component {
+  loadMoreKudos = () => {
+    if (AppStore.kudosStore.canLoadMore && !AppStore.kudosStore.isFetchingKudos) {
+      AppStore.kudosStore.fetchKudos()
     }
   }
 
   render() {
-    const { id, kudos, likeKudo, unlikeKudo, updateKudo, isFetchingKudos } = this.props
     return (
       <div className="kudos-list__container">
-        <TabBarContainer />
+        <TabBar />
         <List
-          userId={id}
-          kudos={kudos}
-          likeKudo={likeKudo}
-          unlikeKudo={unlikeKudo}
-          updateKudo={updateKudo}
-          isFetchingKudos={isFetchingKudos}
+          user={AppStore.user}
+          kudos={AppStore.kudosStore.kudos}
+          isFetchingKudos={AppStore.kudosStore.isFetchingKudos}
         />
-        {isFetchingKudos ? <Spinner /> : null}
+        <div className="kudos-list__fetching-container">
+          {AppStore.kudosStore.isFetchingKudos ? (
+            <Spinner />
+          ) : (
+            <LoadMore onClick={this.loadMoreKudos} />
+          )}
+        </div>
```

Comment:
This was showing in the merge conflict, but please include the `<BottomScrollListener />` component please!